roles/keycloak/vars/main.yml

```
@@ -0,0 +1,78 @@
+# Copyright (c) 2022 VEXXHOST, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+_keycloak_helm_values:
+  # Note(okozachenko1203): Mysql vendor is not supported by bitnami helm chart. As a workaround,
+  #                        we have to define jdbc connection string explicitly along side
+  #                        `externalDatabase` helm values.
+  extraEnvVars:
+    - name: KC_TRANSACTION_XA_ENABLED
+      value: "false"
+    - name: KC_DB
+      value: mysql
+    - name: KC_DB_URL
+      value: "jdbc:mysql://{{ openstack_helm_endpoints.oslo_db.hosts.default }}.openstack:3306/{{ keycloak_database_name }}"
+  auth:
+    adminPassword: "{{ keycloak_admin_password }}"
+    adminUser: "{{ keycloak_admin_username }}"
+  externalDatabase:
+    host: "{{ openstack_helm_endpoints.oslo_db.hosts.default }}.openstack"
+    port: 3306
+    database: "{{ keycloak_database_name }}"
+    existingSecret: "keycloak-db-credentials"
+    existingSecretUserKey: username
+    existingSecretPasswordKey: password
```

we can replace this by using `externalDatabase.user` and `externalDatabase.password`

```suggestion
    user: "{{ keycloak_database_username }}"
    password: "{{ keycloak_database_password }}"
```

I know I suggested auto-generating earlier, but I think it would be better that we do not auto-generate this stuff and keep it for the user to create (the password at least).  That way, the user can rotate passwords if they want easily.